File: projects/Янчук/4/LABB4NEW4/app/src/main/java/com/example/labb4new4/CartAdapter.java
package com.example.labb4new4;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

public class CartAdapter extends BaseAdapter {

    private Context context;
    private List<Product> products;

    public CartAdapter(Context context, List<Product> products) {
        this.context = context;
        this.products = products;
    }

    @Override
    public int getCount() { return products.size(); }

    @Override
    public Object getItem(int i) { return products.get(i); }

    @Override
    public long getItemId(int i) { return products.get(i).id; }

    @Override
    public View getView(int i, View convertView, ViewGroup parent) {
        View v = convertView;
        if (v == null)
            v = LayoutInflater.from(context).inflate(R.layout.item_cart, parent, false);

        Product p = products.get(i);

        TextView tvName = v.findViewById(R.id.tvName);
        TextView tvPrice = v.findViewById(R.id.tvPrice);

        tvName.setText(p.name);
        tvPrice.setText(p.price + " $");

        return v;
    }
}

File: projects/Гончеренок/lab4/src/MiniShop/app/src/main/java/com/example/minishop/CartAdapter.kt
package com.example.minishop

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

class CartAdapter(
    private val context: Context,
    private val items: List<Product>
) : BaseAdapter() {

    override fun getCount(): Int = items.size

    override fun getItem(position: Int): Product = items[position]

    override fun getItemId(position: Int): Long = items[position].id.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.item_cart_product, parent, false)

        val tvId = view.findViewById<TextView>(R.id.tvCartProductId)
        val tvName = view.findViewById<TextView>(R.id.tvCartProductName)
        val tvPrice = view.findViewById<TextView>(R.id.tvCartProductPrice)

        val item = items[position]

        tvId.text = "ID: ${item.id}"
        tvName.text = item.name
        tvPrice.text = "Цена: %.2f руб.".format(item.price)

        return view
    }
}